src/mpi-base.jl

```diff
@@ -529,12 +529,48 @@ function Reduce{T}(object::T, op::Op, root::Integer, comm::Comm)
     isroot ? recvbuf[1] : nothing
 end
 
-function Scatter{T}(sendbuf::MPIBuffertype{T},
-                    count::Integer, root::Integer, comm::Comm)
+function Allreduce{T}(sendbuf::MPIBuffertype{T}, recvbuf::MPIBuffertype{T},
```

Comment:
Julia functions that modify an argument by convention have an exclamation mark appended, as in `Recv!`. The name should be `Allreduce!`.